docs/sources/flow/setup/deploy-agent.md

```diff
@@ -0,0 +1,117 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/setup/deploy-agent/
+canonical: https://grafana.com/docs/agent/latest/flow/setup/start-agent/
+description: Learn about possible deployment topologies for Grafana Agent
+menuTitle: Deploying Grafana Agent
+title: Grafana Agent deployment topologies
+weight: 900
+---
+
+# Deploying Grafana Agent
+
+Grafana Agent is a flexible, vendor-neutral telemetry collector. This
+flexibility means that the Agent doesn’t enforce a specific deployment topology
+but can work in multiple scenarios.
+
+This page lists common topologies used for deployments of Grafana Agent, when
+to consider using each, issues that users may run into, as well as scaling
+considerations.
+
+## As a centralized collection service
+Deploying the Grafana Agent as a centralized service is recommended for
+collecting application telemetry. This allows a smaller set of agents to
+coordinate service discovery, collection, and remote writing.
+
+Using this topology requires deploying the Agent on separate infrastructure,
+and making sure that agents can discover and reach these applications over the
+network. The main predictor for the size of the agent is the number of active
+metrics series it is scraping; a rule of thumb is ~10kb of memory for each
+series. We recommend you start looking towards horizontal scaling around the 1M
+active series mark.
+
+### Using Kubernetes StatefulSets
+Deploying Grafana Agent as a StatefulSet is the recommended option for metrics
+collection.
+The persistent pod identifiers make it possible to consistently match volumes
+with pods so that you can use them for the WAL directory.
+
+You can also use a Kubernetes deployment in cases where persistent storage is not required, such as a traces-only pipeline.
+
+### Pros
+* Straightforward scaling using clustering or hashmod sharding
+* Minimizes the “noisy neighbor” effect
+* Easy to meta-monitor
+
+### Cons
+* Requires running on separate infrastructure
+
+### Use for
+* Scalable telemetry collection
+
+### Don’t use for
+* Host-level metrics and logs
+
+## As a host daemon
+Deploying one Grafana Agent per machine is required for collecting
+machine-level metrics and logs, such as node_exporter hardware and network
+metrics or journald system logs.
+
+Each Agent requires opening an outgoing connection for each remote endpoint
+it’s shipping data to. This can lead to NAT port exhaustion on the egress
+infrastructure. Each egress IP can support up to (65535 - 1024 = 64511)
+outgoing connections on different ports. So, if all agents are shipping metrics
+and log data, an egress IP can support up to 32,255 agents.
```

Comment:
Can someone double-check this here?

Reply:
I don't think its an exact number because you also have any other traffic. Plus if telemetry is enabled that will consume available ports.

Reply:
Yeah, all we can do here is add an upper limit so that people are aware of its effect on their infrastructure.